src/Lib.P2P/Multiplex/Header.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using MultiFormats;
+
+namespace Lib.P2P.Multiplex
+{
+    /// <summary>
+    ///   The header of a multiplex message.
+    /// </summary>
+    /// <remarks>
+    ///   The header of a multiplex message contains the <see cref="StreamId"/> and
+    ///   <see cref="PacketType"/> encoded as a <see cref="Varint">variable integer</see>.
+    /// </remarks>
+    /// <seealso href="https://github.com/libp2p/mplex"/>
+    public struct Header
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Implement 'IEquatable<T>' in value type 'Header'.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)